HomeTask2_2/DiagonalMatrix.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+
+namespace HomeTask2_2;
+
+public class DiagonalMatrix
+{
+    private readonly int Size;
+    public int[] _diagonalElement;
+
+    public DiagonalMatrix(params int[] diagonalElement)
+    {
+        if (diagonalElement == null)
+        {
+            Size = 0;
+            _diagonalElement = new int[0];
+        }
+        else
+        {
+            Size = diagonalElement.Length;
+            _diagonalElement = new int[Size];
+            Array.Copy(diagonalElement, _diagonalElement, Size);
+        }
+    }
+
+    public int this[int i, int j]
+    {
+        get
+        {
+            if (i >= Size || j >= Size || i < 0 || j < 0)
+            {
+                return 0;
+            }
+            return (i == j) ? _diagonalElement[i] : 0;
+        }
+        set
+        {
+            if (i >= 0 && i < Size && j >= 0 && j < Size)
+            {
+                _diagonalElement[i] = value;
+            }
+        }
+    }
+
+    public int Track()
+    {
+        int sum = 0;
+
+        for (int i = 0; i < Size; i++)
+        {
+            sum += _diagonalElement[i];
+        }
+
+        return sum;
+    }
+
+    public override bool Equals(object? obj)
+    {
```

Comment:
Better  if (!(obj is DiagonalMatrix m) || Size != m.Size)
 {
     return false;
 }